routers/repo/issue_dependency_remove.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package repo
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/context"
+)
+
+// RemoveDependency removes the dependency
+func RemoveDependency(c *context.Context) {
+	depID, err := strconv.ParseInt(c.Req.PostForm.Get("removeDependencyID"), 10, 64)
+	if err != nil {
+		c.Handle(http.StatusBadRequest, "issue ID is not int", err)
+		return
+	}
+
+	issueIndex := c.ParamsInt64("index")
+	issue, err := models.GetIssueByIndex(c.Repo.Repository.ID, issueIndex)
+	if err != nil {
+		c.Handle(http.StatusInternalServerError, "GetIssueByIndex", err)
+		return
+	}
+
+	// Dependency Type
+	// Types: 1 = blockedBy, 2 = blocking
+	depType, err := strconv.ParseInt(c.Req.PostForm.Get("dependencyType"), 10, 64)
```

Comment:
You can use map. Eg:
```
depTypes := map[int]DependencyType{1: DependencyTypeBlockedBy, 2: DependencyTypeBlocking}
```

and check:

```
depType, ok := depTypes[c.QueryInt("dependencyType")]
if !ok {
	c.Handle(http.StatusBadRequest, "GetDependecyType", nil)
	return
}
```

Reply:
like the idea, but wouldn't this mean i had to define the dependency types two times? (which i'd rather want to avoid)

Reply:
I think no. This is mapping. Actualy I think you can change it from `int` to `string` and use strings in post calls (as it would be more descriptive). Something like:
```
map[int]DependencyType{"blockedby": DependencyTypeBlockedBy, "blocking": DependencyTypeBlocking}
```
You know that `DependencyTypeXYZ` is also int, but in reality it is custom type that can be anything. If you use constnants you don't care about real type (it can be `type DependencyType string` for example). Also it's not same number:
```
const (
	DependencyTypeBlockedBy DependencyType = iota    // = 0
	DependencyTypeBlocking                           // = 1
)
```

Reply:
Solved it without a map.